core/src/main/java/org/apache/gravitino/authorization/OwnershipManager.java

```diff
@@ -0,0 +1,138 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.gravitino.Entity;
+import org.apache.gravitino.EntityStore;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.exceptions.NoSuchEntityException;
+import org.apache.gravitino.exceptions.NotFoundException;
+import org.apache.gravitino.exceptions.OwnerNotFoundException;
+import org.apache.gravitino.meta.GroupEntity;
+import org.apache.gravitino.meta.UserEntity;
+import org.apache.gravitino.relation.Relation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** OwnershipManager is used for manage the owner of securable objects, user, group and role. */
+public class OwnershipManager {
+  private static final Logger LOG = LoggerFactory.getLogger(OwnershipManager.class);
+  private final EntityStore store;
+
+  public OwnershipManager(EntityStore store) {
+    this.store = store;
+  }
+
+  public void setOwner(
+      NameIdentifier identifier, Entity.EntityType type, String ownerName, Owner.Type ownerType) {
+    try {
+      String metalake;
+      if (type == Entity.EntityType.METALAKE) {
+        metalake = identifier.name();
+      } else {
+        metalake = identifier.namespace().level(0);
+      }
+
+      if (ownerType == Owner.Type.USER) {
+        store
+            .relationOperations()
+            .insertRelation(
+                Relation.Type.OWNER_REL,
+                identifier,
+                type,
+                AuthorizationUtils.ofUser(metalake, ownerName),
+                Entity.EntityType.USER,
+                true);
+      } else if (ownerType == Owner.Type.GROUP) {
+        store
+            .relationOperations()
+            .insertRelation(
+                Relation.Type.OWNER_REL,
+                identifier,
+                type,
+                AuthorizationUtils.ofGroup(metalake, ownerName),
+                Entity.EntityType.GROUP,
+                true);
+      }
+    } catch (NoSuchEntityException nse) {
+      LOG.warn("Entity {} or owner {} is not found", identifier, ownerName, nse);
+      throw new NotFoundException(nse, "Entity %s or owner %s is not found", identifier, ownerName);
+    } catch (IOException ioe) {
+      LOG.info("Fail to set the owner {} of entity {}", ownerName, identifier, ioe);
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  public Owner getOwner(NameIdentifier identifier, Entity.EntityType type) {
+    try {
+      OwnerImpl owner = new OwnerImpl();
+      List entities =
+          store
+              .relationOperations()
+              .listEntitiesByRelation(Relation.Type.OWNER_REL, identifier, type);
+      if (entities.isEmpty()) {
```

Comment:
I'm afriad that existing entities may has no ower and then throw an exception here.

Reply:
Yes, it will throw an exception. We should return the exception to the client, too.